setup.ts

```diff
@@ -0,0 +1,379 @@
+    for (const key in config) {
+      fs.appendFileSync(".env", `${key}=${config[key]}\n`);
+    }
+  }
+}
+
+// Check the connection to MongoDB with the specified URL.
+async function checkConnection(url: string): Promise<boolean> {
+  let response = false;
+  const client = new mongodb.MongoClient(url, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+    serverSelectionTimeoutMS: 1000,
+  });
+
+  console.log("\nChecking MongoDB connection....");
+
+  try {
+    await client.connect();
+    console.log("\nConnection to MongoDB successful! 🎉");
+    response = true;
+  } catch (error) {
+    console.log(`\nConnection to MongoDB failed. Please try again.\n`);
+  }
+  client.close();
+  return response;
+}
+
+//Mongodb url prompt
+async function askForMongoDBUrl(): Promise<string> {
+  const { url } = await inquirer.prompt([
+    {
+      type: "input",
+      name: "url",
+      message: "Enter your MongoDB URL:",
+    },
+  ]);
+
+  return url;
+}
+
+// Get the mongodb url
+async function mongoDB(): Promise<void> {
+  let DB_URL = process.env.MONGO_DB_URL;
+
+  try {
+    let isConnected = false,
+      url = "";
+    while (!isConnected) {
+      url = await askForMongoDBUrl();
+      isConnected = await checkConnection(url);
+    }
+
+    DB_URL = url;
+    const config = dotenv.parse(fs.readFileSync(".env"));
+    config.MONGO_DB_URL = DB_URL;
+    process.env.MONGO_DB_URL = DB_URL;
+    fs.writeFileSync(".env", "");
+    for (const key in config) {
+      fs.appendFileSync(".env", `${key}=${config[key]}\n`);
+    }
+  } catch (err) {
+    console.error(err);
+    abort();
+  }
+}
+
+//Get recaptcha details
+async function recaptcha(): Promise<void> {
+  console.log(
+    "\nPlease visit this URL to set up reCAPTCHA:\n\nhttps://www.google.com/recaptcha/admin/create"
+  );
+  console.log(
+    '\nSelect reCAPTCHA v2 and the "I`m not a robot" checkbox option'
+  );
+  console.log(
+    '\nAdd "localhost" in domains and accept the terms, then press submit'
+  );
+
+  const { recaptchaSecretKey } = await inquirer.prompt([
+    {
+      type: "input",
+      name: "recaptchaSecretKey",
+      message: "Enter your reCAPTCHA secret site key:",
+      validate: async (input: string): Promise<boolean | string> => {
+        if (validateRecaptcha(input)) {
+          return true;
+        }
+        return "Invalid reCAPTCHA secret key. Please try again.";
+      },
+    },
+  ]);
+  const config = dotenv.parse(fs.readFileSync(".env"));
+  config.RECAPTCHA_SECRET_KEY = recaptchaSecretKey;
+  fs.writeFileSync(".env", "");
+  for (const key in config) {
+    fs.appendFileSync(".env", `${key}=${config[key]}\n`);
+  }
+}
+
+function isValidEmail(email: string): boolean {
+  const pattern = /^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/;
+  const match = email.match(pattern);
+  return match !== null && match[0] === email;
+}
+
+function validateRecaptcha(string: string): boolean {
+  const pattern = /^[a-zA-Z0-9_-]{40}$/;
```

Comment:
make sure this regex is valid, if it is ignore the comment

Reply:
Yes, as far as I have tested, it is correct